### PR TITLE
Testing fixes

### PR DESCRIPTION
**Pull request by jedwards4b**
Fixes issues with LII and ERP tests.   Improves specification of _P modifier in test names.   Fixes an issue with shared build conflicts between clm and dlnd.  Fixes an issue with cime_bisect unit test. 

Test suite: scripts_regression_tests on yellowstone and hobart
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes: 

User interface changes?: 

Code review: jgfouca

### CODE REVIEW

File: scripts/create_newcase [hunk 1]
```diff
@@ -52,8 +52,9 @@ def parse_command_line(args, cimeroot):
     parser.add_argument("--project", "-project",
                         help="Specify a project id")
 
-    parser.add_argument("--pecount", "-pecount", choices=('S','M','L','X1','X2'), default="M",
-                        help="Specify a target size description for the number of cores")
+    parser.add_argument("--pecount", "-pecount",  default="M",
```
Comment (jedwards4b) [inline scripts/create_newcase]: Needed to add support for the TASKSxCOUNT option here.  It was previously handled in create_test after create_newcase was completed, but I ran into a problem where the default pelayout was too big for the test system which caused create_newcase to throw an error before the pelayout adjustment specified in the test could be applied.    The choices option to add_argument was a nice feature, but I couldn't figure out how to use it with these additional possibilities.

File: scripts/create_test [hunk 2]
```diff
@@ -161,6 +161,8 @@ OR
                         help="Use this category key in the lookup in testlist.xml")
     parser.add_argument("--xml-testlist",
                         help="Use this testlist to lookup tests")
+    parser.add_argument("--testfile", nargs=1, default=None,
```
Comment (jedwards4b) [inline scripts/create_test]: This allows the user to specify multiple tests in a file rather than on the command line.

File: scripts/create_test [hunk 3]
```diff
@@ -327,12 +336,6 @@ def create_test(testargs, compiler, machine_name, no_run, no_build, no_batch, te
                 else:
                     expect(machine_name == testsplit[4],
                            "ambiguity in machine, please use the --machine option")
-            if testsplit[5] is not None:
```
Comment (jedwards4b) [inline scripts/create_test]: We were unnecessarily restricting all tests to use the same compiler.   No such restriction should be required.

File: utils/python/CIME/SystemTests/erp.py [hunk 4]
```diff
@@ -142,7 +142,11 @@ def run(self):
             if not success:
                 break
 
-        return success
+        if success:
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/erp.py]: Fixes a bug in ERP - restart was not being compared to initial run.

File: utils/python/CIME/SystemTests/lii.py [hunk 5]
```diff
@@ -43,49 +43,38 @@ def build(self, sharedlib_only=False, model_only=False):
             for filename in glob.glob(r'user_nl_clm*'):
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/lii.py]: The lii test is very specific to the clm model and compares an initial run from internally interpolated initial data to one from externally interpolated initial data.

File: utils/python/CIME/SystemTests/system_tests_common.py [hunk 6]
```diff
@@ -280,7 +280,7 @@ def compare_baseline(self):
         if not os.path.isfile(baselog):
             # for backward compatibility
             baselog = os.path.join(basecmp_dir, "cpl.log")
-        if len(memlist) > 3:
+        if baselog is not None and len(memlist) > 3:
```
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/system_tests_common.py]: Avoid an error here if baselog does not exist.

File: utils/python/CIME/XML/env_batch.py [hunk 7]
```diff
@@ -277,6 +277,7 @@ def set_job_defaults(self, bjobs, pesize=None, walltime=None):
             self.set_value("JOB_QUEUE", queue, subgroup=job)
             walltime = self.get_max_walltime(queue) if walltime is None else walltime
             if walltime is None:
+                logger.warn("Could not find a queue matching task count %d, falling back to depreciated default walltime parameter"%task_count)
```
Comment (jedwards4b) [inline utils/python/CIME/XML/env_batch.py]: Improve the error message here.

File: utils/python/CIME/build.py [hunk 8]
```diff
@@ -511,7 +511,7 @@ def build_libraries(case, exeroot, caseroot, cimeroot, libroot, mpilib, lid, mac
         sharedpath = os.environ["SHAREDPATH"]
         bldroot = os.path.join(sharedpath, case.get_value("COMP_INTERFACE"), esmfdir, "clm","obj" )
         libroot = os.path.join(sharedpath, case.get_value("COMP_INTERFACE"), esmfdir, "lib")
-        incroot = os.path.join(sharedpath,"include")
+        incroot = os.path.join(sharedpath, case.get_value("COMP_INTERFACE"), esmfdir, "include")
```
Comment (jedwards4b) [inline utils/python/CIME/build.py]: There was a conflict between the dlnd lnd_comp_mct.mod and the clm lnd_comp_mct.mod - moving the location of the clm include directory solves this problem.

File: utils/python/CIME/case.py [hunk 9]
```diff
@@ -538,11 +538,36 @@ def configure(self, compset_name, grid_name, machine_name=None,
         #--------------------------------------------
         # pe payout
         #--------------------------------------------
-        pesobj = Pes(self._pesfile)
+        match1 = False
```
Comment (jedwards4b) [inline utils/python/CIME/case.py]: Moved here from system_test.py

File: utils/python/CIME/utils.py [hunk 10]
```diff
@@ -322,7 +322,7 @@ def get_full_test_name(partial_test, grid=None, compset=None, machine=None, comp
             expect(arg_val is not None,
                    "Could not fill-out test name, partial string '%s' had no %s information and you did not provide any" % (partial_test, name))
             result = "%s%s%s" % (result, "_" if name == "compiler" else ".", arg_val)
-        elif (arg_val is not None):
+        elif (arg_val is not None and partial_val != partial_compiler):
```
Comment (jedwards4b) [inline utils/python/CIME/utils.py]: Avoid error if tests use different compilers.

File: utils/python/tests/scripts_regression_tests.py [hunk 11]
```diff
@@ -39,8 +39,8 @@ def do_unit_tests(self, script,from_dir=SCRIPT_DIR):
         stat, output, _ = run_cmd("./%s --test 2>&1" % script, from_dir=from_dir)
         self.assertEqual(stat, 0, msg=output)
 
-    def test_acme_bisect_unit_test(self):
-        self.do_unit_tests("acme_bisect",from_dir=TOOLS_DIR)
+    def test_cime_bisect_unit_test(self):
```
Comment (jedwards4b) [inline utils/python/tests/scripts_regression_tests.py]: update to match script name change

**Comment by jedwards4b** [comment]
annotation complete

**Comment by billsacks** [comment]
@jedwards4b : Regarding the LII test: From a read-through of the code, it looks like you fixed some problems but introduced a different problem: both runs call

```python
            success = SystemTestsCommon._run(self)
```

without giving the optional suffix argument. I'm just coming to understand the test scripts, but I'm pretty sure that's a problem.

Also: I found the previous style, with explicit methods to do the first and second runs, more understandable.

Since I already having fixing the LII test and doing some rigorous testing of it on my task list: Can you please simply back out your changes to the LII test here? Then I'll submit a separate PR with a fixed LII test.

File: scripts/create_test (hunk 2 above)
Comment (jgfouca) [inline scripts/create_test]: nargs=1 and default=None are both default values for a string option, they don't need to be specified.

File: scripts/create_test (hunk 3 above)
Comment (jgfouca) [inline scripts/create_test]: I thought we already fixed this. Did we just forget to change this particular block?

File: utils/python/CIME/SystemTests/system_tests_common.py (hunk 6 above)
Comment (jgfouca) [inline utils/python/CIME/SystemTests/system_tests_common.py]: It's not possible for baselog to be None here due to line 279.

File: utils/python/CIME/case.py [hunk 12]
```diff
@@ -538,11 +538,36 @@ def configure(self, compset_name, grid_name, machine_name=None,
...
+                string = "NTHRDS_" + component_class
+                pes_nthrds[string] = opti_thrds
+                string = "ROOTPE_" + component_class
+                pes_rootpe[string] = 0
+        else:
+            pesobj = Pes(self._pesfile)
 
-        #FIXME - add pesize_opts as optional argument below
-        pes_ntasks, pes_nthrds, pes_rootpe = pesobj.find_pes_layout(self._gridname, self._compsetname,
+            pes_ntasks, pes_nthrds, pes_rootpe = pesobj.find_pes_layout(self._gridname, self._compsetname,
```
Comment (jgfouca) [inline utils/python/CIME/case.py]: Was this indent intentional? This means that find_pes_layout is only run in the else block.

Comment (jgfouca) [inline utils/python/CIME/case.py]: Nevermind, I think I understand what you did here.

**Comment by jgfouca** [comment]
@jedwards4b : A couple minor changes needed and then this is ready pending resolution of @billsacks comment.

File: scripts/create_test (hunk 2 above)
Comment (jedwards4b) [inline scripts/create_test]: fixed

File: scripts/create_test (hunk 3 above)
Comment (jedwards4b) [inline scripts/create_test]: I fixed it for xml tests but not for tests listed on the command line or in a testfile.

File: utils/python/CIME/SystemTests/system_tests_common.py (hunk 6 above)
Comment (jedwards4b) [inline utils/python/CIME/SystemTests/system_tests_common.py]: good point, changed to os.path.isfile(baselog)

**Pull request merged**